server/plugins/openAiDallE3Plugin.js

```diff
@@ -52,14 +52,20 @@ class OpenAIDallE3Plugin extends ModelPlugin {
 
         requestPromise
         .then((response) => handleResponse(response))
-        .catch((error) => handleResponse(error));
+        .catch((error) => handleResponse(error, true));
 
-        function handleResponse(response) {
+        function handleResponse(response, isError = false) {
             let status = "succeeded";
-            let data = JSON.stringify(response);
-            if (response.data.error) {
+            let data;
+
+            if (isError) {
+                status = "failed";
+                data = JSON.stringify({ error: response.message || response });
```

Comment:
merge the two error ifs? makes any sense?

Reply:
Could do this:

`            const status = isError || response.data?.error ? "failed" : "succeeded";
            const data = JSON.stringify(
                isError ? { error: response.message || response } :
                response.data?.error ? response.data :
                response
            );`

But I don't really want to read that, do you? :D